src/basic/env-file.c

```diff
@@ -225,10 +211,13 @@ static int parse_env_file_internal(
                 case DOUBLE_QUOTE_VALUE_ESCAPE:
                         state = DOUBLE_QUOTE_VALUE;
 
-                        if (!strchr(NEWLINE, c)) {
+                        if (c == '\"')
+                                value[n_value++] = c;
```

Comment:
You need to do the GREEDY_REALLOC in this case as well.

I recommend something like:

```c
if (c == '\"' || !strchr(NEWLINE, c)) {
    if (!GREEDY_REALLOC(value, value_alloc, n_value+3))
        return -ENOMEM;
    if (c != '\"')
        value[n_value++] = '\\';
    value[n_value++] = c;
}
```

(Untested, might be incorrect...)

Reply:
A GREEDY_REALLOC is needed, but I don't think the proposed code is good.
Maybe just
```C 
if (c == '"') {
        if (!GREEDY_REALLOC(value, value_alloc, n_value+2))
                return -ENOMEM;
        value[n_value++] = '"';
} else if (!strchr(NEWLINE, c)) {
       if (!GREEDY_REALLOC(value, value_alloc, n_value+3))
                 return -ENOMEM;
       value[n_value++] = '\\';
       value[n_value++] = c;
}

Reply:
Good point, I'll make this change.